dipy/denoise/gibbs.py

```diff
@@ -262,33 +271,40 @@ def gibbs_removal(vol, slice_axis=2, n_points=3, inplace=True):
     """
     nd = vol.ndim
 
+    # check matrix dimension
+    if nd > 4:
+        raise ValueError("Data have to be a 4D, 3D or 2D matrix")
+    elif nd < 2:
+        raise ValueError("Data is not an image")
+
     if not isinstance(inplace, bool):
         raise TypeError("inplace must be a boolean.")
 
+    if not isinstance(num_processes, int):
+        raise TypeError("num_processes must be an int.")
+    else:
+        if num_processes < 0:
+            raise ValueError("num_processes must be >= 0.")
```

Comment:
Can you add new tests for all this block?

Reply:
All tests added.